integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/datamap/CarbonDataMapRebuildCommand.scala

```diff
@@ -48,7 +50,17 @@ case class CarbonDataMapRebuildCommand(
         )(sparkSession)
     }
     val provider = DataMapManager.get().getDataMapProvider(table, schema, sparkSession)
-    provider.rebuild()
+    // for non-lazy index datamap, the data of datamap will be generated immediately after
+    // the datamap is created or the main table is loaded, so there is no need to
+    // rebuild this datamap.
+    if (!schema.isLazy && provider.isInstanceOf[IndexDataMapProvider]) {
```

Comment:
Even if it is other datamap like preaggregat datamap, we should not rebuild it, right?

Reply:
For MV, current implementation requires rebuild.
For preagg, I'm not sure about its implementation, so leave it as it is.

Reply:
OK.